src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1653,6 +1655,37 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
             .containsOnly(REPLICA);
     }
 
+    @Test
```

Comment:
You were close in `shouldKeepUsingReplicaAfterSettingTimeout` test on covering `set statement_timeout` use case. But it wasn't tested from this perspective